### ISSUE TITLE
User doesn't unblock after blocked.

### ISSUE THREAD

**Issue by kishorejethava** [QuickBlox/quickblox-android-sdk#583]
User doesn't unblock after blocked from privacy list 

Written in **kotlin**

    private fun blockFromQuickBlox(qbId: String?) {
        val privacyListsManager = QBChatService.getInstance().privacyListsManager
        privacyListsManager.addPrivacyListListener(object : QBPrivacyListListener {
            override fun setPrivacyList(listName: String?, listItem: MutableList<QBPrivacyListItem>?) {
                Log.i(TAG,"Privacy listName:"+listName +" listItem:"+listItem.toString())
            }

            override fun updatedPrivacyList(listName: String?) {
                Log.i(TAG,"Privacy listName:"+listName)
            }
        })

        try {
            privacyListsManager.declineActiveList()
            privacyListsManager.declineDefaultList()
        } catch (e : SmackException) {
            Log.e(TAG,"SmackException:"+e.printStackTrace())
        } catch (e : XMPPException){
            Log.e(TAG,"XMPPException:"+e.printStackTrace())
        }

        try {
            val privacyList = privacyListsManager.getPrivacyList("public")
            Log.e(TAG,"privacy list:"+privacyList.items.toString())

            privacyList.isActiveList = true
            val listPrivacyListItem = ArrayList<QBPrivacyListItem>()
            val qbPrivacyListItem = QBPrivacyListItem()
            qbPrivacyListItem.type = QBPrivacyListItem.Type.USER_ID
            qbPrivacyListItem.valueForType = qbId
            qbPrivacyListItem.isMutualBlock = false

            listPrivacyListItem.add(qbPrivacyListItem)

            privacyList.items = listPrivacyListItem
            privacyList.isDefaultList = true

            privacyListsManager.setPrivacyList(privacyList)
            privacyListsManager.setPrivacyListAsDefault("public")
            privacyListsManager.setPrivacyListAsActive("public")
            
            } catch (e : SmackException.NotConnectedException) {
                Log.e(TAG,"NotConnectedException:"+e.printStackTrace())
            } catch (e : XMPPException.XMPPErrorException ) {
                Log.e(TAG,"XMPPErrorException:"+e.printStackTrace())
            } catch (e : SmackException.NoResponseException) {
                Log.e(TAG,"NoResponseException:"+e.printStackTrace())
            }
    }`

I have also tried to update existing item

     privacyList.items.forEachIndexed { index, qbPrivacyListItem ->
                Log.i(TAG,"qbPrivacyListItem.valueForType:"+qbPrivacyListItem.valueForType+" qbId:"+qbId)
                if(qbPrivacyListItem.valueForType.contains(qbId!!)){
                    Log.i(TAG,"isMutualBlock:"+qbPrivacyListItem.isMutualBlock) // print false
                    qbPrivacyListItem.isMutualBlock = false
                    Log.i(TAG,"isMutualBlock:"+qbPrivacyListItem.isMutualBlock) // print false

                }
            }

**Comment by RomanPronin** [comment]
Try to use the SDK version 3.8.0 with new declinePrivacyList() logic without declineDefaultList() and declineActiveList().

**Comment by kishorejethava** [comment]
@RomanPronin I can't use latest version because it conflict with other dependency.
Is there any other way because I'm trying from last one month.

**Comment by RomanPronin** [comment]
@kishorejethava, pls take a look at similar https://github.com/QuickBlox/quickblox-android-sdk/issues/580 but use old deprecated methods.

**Comment by QB-nikolay-zolotarov** [comment]
Hello QuickBlox customer,

This is Nikolay from QuickBlox support.

The issue was closed as it was outdated.

Please update the SDK to the latest version:
[https://github.com/QuickBlox/quickblox-android-sdk-releases/releases/tag/3.9.11](https://github.com/QuickBlox/quickblox-android-sdk-releases/releases/tag/3.9.11)

Additionally, please check our new samples:
[https://docs.quickblox.com/docs/code-samples#chat-samples](https://docs.quickblox.com/docs/code-samples#chat-samples)

If it is still relevant after reviewing the updated information, feel free to open a new issue.

Have a nice day.
